modules/contacts/contacts.service.js

```diff
@@ -1,20 +1,66 @@
 const Contact = require('./contacts.model');
+const {
+  CONTACT_NOT_FOUND,
+  CONTACT_ALREADY_EXIST,
+} = require('../../error-messages/contacts.messages');
 
 class ContactsService {
   async getContacts() {
     return await Contact.find();
   }
 
+  async getContactById(id) {
+    const contact = await Contact.findById(id);
+
+    if (!contact) {
+      throw new Error(CONTACT_NOT_FOUND);
+    }
+
+    return contact;
+  }
+
   async addContact(data) {
+    if (await this.checkContactExist(data)) {
+      throw new Error(CONTACT_ALREADY_EXIST);
+    }
+
     return new Contact(data).save();
   }
 
-  async updateContact(id, contact) {
-    return await Contact.findByIdAndUpdate(id, contact);
+  async updateContact(id, contactData) {
+    const contact = await Contact.findByIdAndUpdate(id, contactData);
```

Comment:
You need to add third argument `{ new: true }` to ensure that updated data is returned

Reply:
It is better to not throw an error here but return update contact or empty value it to the resolver and return an error there